cpp/src/Slice/Parser.cpp

```diff
@@ -5037,15 +5060,16 @@ Slice::Unit::getTopLevelModules(const string& file) const
     }
 }
 
-Slice::Unit::Unit(bool all, MetadataList defaultFileMetadata)
+Slice::Unit::Unit(std::string_view languagePrefix, bool all, MetadataList defaultFileMetadata)
     : SyntaxTreeBase(nullptr),
       Container(nullptr),
+      _languagePrefix(languagePrefix),
       _all(all),
       _defaultFileMetadata(std::move(defaultFileMetadata)),
       _errors(0),
       _currentIncludeLevel(0)
-
 {
+    assert(binary_search(&languages[0], &languages[sizeof(languages) / sizeof(*languages)], _languagePrefix));
```

Comment:
How does this work for 'icerpc'?

Reply:
It didn't, but I fixed it. And opened an issue to run `ice2slice` in CI.
Right now, we have 0 testing of this tool, and it's way too easy to break it without realizing...